bob-apps/bin/bobp.rs

```diff
@@ -384,24 +416,39 @@ async fn print_errors_with_codes(stat: Arc<Statistics>) {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
+    let guard = stat.exist_errors.lock().await;
+    if !guard.is_empty() {
+        println!("put errors:");
+        for (&code, count) in guard.iter() {
+            println!("{:?} = {}", Code::from(code), count);
+        }
+    }
 }
 
 fn print_averages(
     stat: &Arc<Statistics>,
     put_speed_values: &[f64],
     get_speed_values: &[f64],
+    exist_speed_values: &[f64],
     elapsed: Duration,
 ) {
+    let avg_total = ((stat.put_total.load(Ordering::Relaxed) + 
+                    stat.get_total.load(Ordering::Relaxed) +
+                    stat.exist_total.load(Ordering::Relaxed)) * 1000)
+                    .checked_div(elapsed.as_millis() as u64)
+                    .unwrap_or_default();
+    let total_err = stat.put_error_count.load(Ordering::Relaxed) + 
+                    stat.get_error_count.load(Ordering::Relaxed) +
+                    stat.exist_error_count.load(Ordering::Relaxed);
     println!(
         "avg total: {} rps | total err: {}\r\n\
-        put: {:>6.2} kb/s | get: {:>6.2} kb/s\r\n\
-        put resp time, ms: {:>6.2} | get resp time, ms: {:>6.2}",
-        ((stat.put_total.load(Ordering::Relaxed) + stat.get_total.load(Ordering::Relaxed)) * 1000)
-            .checked_div(elapsed.as_millis() as u64)
-            .unwrap_or_default(),
-        stat.put_error_count.load(Ordering::Relaxed) + stat.get_error_count.load(Ordering::Relaxed),
+        put: {:>6.2} kb/s | get: {:>6.2} kb/s | exist: {:>6.2} kb/s\r\n\
+        put resp time, ms: {:>6.2} | get resp time, ms: {:>6.2} | exist resp time, ms: {:>6.2}",
```

Comment:
Same here: it is better to print information only about operations that was actually executed